README.md

```diff
@@ -145,378 +122,349 @@ const [new_user] = await sql`
-
-This also means you cannot write dynamic queries or concat queries together by simple string manipulation. To enable dynamic queries in a safe way, the `sql` function doubles as a regular function which escapes any value properly. It also includes overloads for common cases of inserting, selecting, updating and querying.
+## Dynamic queries
 
-## Dynamic query helpers - `sql()` inside tagged template
+Postgres.js features a powerful dynamic query parser for conditionally appending/omitting query fragments.
 
-Postgres.js has a safe, ergonomic way to aid you in writing queries. This makes it easier to write dynamic `insert`, `select` and `update` queries, and pass `where` parameters.
+This works by nestings a ` sql`` ` call within another ` sql`` ` call.
 
-#### Insert
+#### Partial queries
 
 ```js
+let savedQuery = () => sql`and age > 50`
 
-const user = {
-  name: 'Murray',
-  age: 68
-}
+let isQueryingForAge = true
 
 sql`
-  insert into users ${
-    sql(user, 'name', 'age')
-  }
+  select
+   *
+  from users 
+  where 
+    name is not null
+    ${isQueryingForAge ?
+     savedQuery()
+     :
+     sql``
+    }
 `
-
-// Is translated into this query:
-insert into users ("name", "age") values ($1, $2)
-
 ```
 
-You can leave out the column names and simply do `sql(user)` if you want to get all fields from the object as columns, but be careful not to allow users to supply columns you don't want.
+#### Dynamic where clause
+```js
+sql` 
+  select 
+    * 
+  from users ${id ? 
+    sql`where user_id = ${ id }` 
+    : 
+    sql`` 
+  }
+`
 
-#### Multiple inserts in one query
-If you need to insert multiple rows at the same time it's also much faster to do it with a single `insert`. Simply pass an array of objects to `sql()`.
+// Is translated to:
+select * from users
+// Or
+select * from users where user_id = $1
+```
 
+#### Dynamic filters
 ```js
+let ageFilter = 50;
 
-const users = [{
-  name: 'Murray',
-  age: 68,
-  garbage: 'ignore'
-}, {
-  name: 'Walter',
-  age: 78
-}]
-
-sql`
-  insert into users ${
-    sql(users, 'name', 'age')
+sql` 
+  select 
+    * 
+  from users 
+  where
+   age > ${ageFilter}
+  ${id ? 
+    sql`and user_id = ${id}` 
+    : 
+    sql`` 
   }
 `
 
-// Is translated into this query:
-insert into users ("name", "age") values ($1, $2), ($3, $4)
+// Is translated to:
+select * from users where age > $1
+// Or
+select * from users where age > $1 and user_id = $2
 ```
 
-#### Update
+### Identifier and value utilities
 
-This is also useful for update queries 
-```js
+#### Arrays
+Arrays will be handled by replacement parameters too, so `where in` queries are also simple.
 
-const user = {
-  id: 1,
-  name: 'Muray'
-}
+```js
 
-sql`
-  update users set ${
-    sql(user, 'name')
-  } where 
-    id = ${ user.id }
+const users = await sql`
+  select
+    *
+  from users
+  where age in (${ [68, 75, 23] })
```

Comment:
This is deprecated in v3. It's done like this now by removing parenthesis and using sql():
```
in ${ sql(array) }
```